x-pack/test_serverless/api_integration/services/slo_api.ts

```diff
@@ -148,6 +152,23 @@ export function SloApiProvider({ getService }: FtrProviderContext) {
       });
     },
 
+    async waitForSloReseted(sloId: string, roleAuthc: RoleCredentials) {
+      if (!sloId) {
+        throw new Error('sloId is undefined');
+      }
+      return await retry.tryForTime(retryTimeout, async () => {
+        const response = await supertestWithoutAuth
+          .post(`/api/observability/slos/${sloId}/_reset`)
+          .set(svlCommonApi.getInternalRequestHeader())
+          .set(roleAuthc.apiKeyHeader)
+          .timeout(requestTimeout);
```

Comment:
Feel free to ignore, just sharing another way to handle the wait: 

If it makes sense to "fail fast" and not slow the test because of high request timeout (30s), the alternative is to set request retry count and decrease timeout. E.g. 10s request timeout and 2s delay before the next attempt

```
      return await retry.tryWithRetries(
        `Wait for SLO to be created`,
        async () => {
          const response = await supertestWithoutAuth
            .get(`/api/observability/slos/${sloId}`)
            .set(svlCommonApi.getInternalRequestHeader())
            .set(roleAuthc.apiKeyHeader)
            .timeout(10000);
          if (response.body.id === undefined) {
            throw new Error(`No slo with id ${sloId} found`);
          }
          return response.body;
        },
        { retryCount: 5, retryDelay: 2000 }
      );
```

Alerting API services uses it https://github.com/elastic/kibana/blob/main/x-pack/test/api_integration/deployment_agnostic/services/alerting_api.ts#L97-L130